rest_framework/fields.py

```diff
@@ -1270,6 +1274,70 @@ def to_internal_value(self, data):
         return data
 
 
+class RecursiveField(Field):
+    """
+    A field that gets its representation from its parent.
+
+    This method could be used to serialize a tree structure, a linked list, or
+    even a directed acyclic graph. As with all recursive things, it is
+    important to keep the base case in mind. In the case of the tree serializer
+    example below, the base case is a node with an empty list of children. In
+    the case of the list serializer below, the base case is when `next==None`.
+    Above all, beware of cyclical references.
+
+    Examples:
+
+    class TreeSerializer(self):
+        children = ListField(child=RecursiveField())
+
+    class ListSerializer(self):
+        next = RecursiveField(allow_null=True)
+    """
+
+    def __init__(self, to='self', to_module=None, **kwargs):
```

Comment:
I wanted to enable capture the case where the serializer isn't necessarily defined in the top-level namespace of the module (in case it was defined nested within another class?). This is a very border use case so I'd be fine ignoring it.